src/vmod_dynamic.c

```diff
@@ -727,7 +727,9 @@ dynamic_get(VRT_CTX, struct vmod_dynamic_director *obj, const char *addr,
 	AZ(pthread_cond_init(&dom->cond, NULL));
 	AZ(pthread_cond_init(&dom->resolve, NULL));
 
-	AZ(pthread_create(&dom->thread, NULL, &dynamic_lookup_thread, dom));
+	if (ctx->method != VCL_MET_INIT)
+		AZ(pthread_create(&dom->thread, NULL, &dynamic_lookup_thread,
```

Comment:
I pondered this for a bit, but actually I think now the solution can be so simple.
Alternatively, we could synchronize in `dynamic_lookup_thread()`, allowing multiple lookup threads to be started (so all but one would die off again), but I do not see a case for this complication at the moment.